src/display/text_layer.js

```diff
@@ -426,9 +447,60 @@ function renderTextLayer(renderParameters) {
     viewport: renderParameters.viewport,
     textDivs: renderParameters.textDivs,
     textContentItemsStr: renderParameters.textContentItemsStr,
+    textDivProperties: renderParameters.textDivProperties,
+    isOffscreenCanvasSupported: renderParameters.isOffscreenCanvasSupported,
   });
   task._render();
   return task;
 }
 
-export { renderTextLayer, TextLayerRenderTask };
+/**
+ * @param {TextLayerRenderParameters} renderParameters
+ * @returns {TextLayerRenderTask}
+ */
+function updateTextLayer(renderParameters) {
+  const { isOffscreenCanvasSupported, textDivProperties, textDivs, viewport } =
+    renderParameters;
+  const ctx = getCtx(0, isOffscreenCanvasSupported);
+  const scale = viewport.scale * (globalThis.devicePixelRatio || 1);
+  const params = {
+    prevFontSize: null,
+    prevFontFamily: null,
+    div: null,
+    scale,
+    properties: null,
+    ctx,
+  };
+  for (const div of textDivs) {
+    params.properties = textDivProperties.get(div);
+    params.div = div;
+    layout(params);
+  }
+}
+
+/**
+ * @param {HTMLDivElement} div
+ * @param {import("./display_utils").PageViewport} viewport
+ */
+function setTextLayerDimensions(div, viewport) {
```

Comment:
It's because we pass to the render a `Fragment` where to add the spans and not the div itself.
It was already like that before the patch and the div is currently getting its dimensions in the `pdf_page_view`:
https://github.com/mozilla/pdf.js/blob/fa54a587906100eb8198de2e95d6b1e0bfa88846/web/pdf_page_view.js#L704-L705
So I'm not sure it's really a regression.
Anyway, I already thought to get rid of the `Fragment` but I thought doing that later to avoid to add more complexity to this patch (I'm not joking :))
As said before, I think we really should refactor the layer stuff (text, annotation, editor and xfa) to have something more consistent.

Reply:
I removed the fragment and pass the div to the renderer.

Reply:
> It was already like that before the patch and the div is currently getting its dimensions in the `pdf_page_view`:

Sure, but things become even worse now (as evident by the changes in `test/driver.js`) hence my questions here.

> Anyway, I already thought to get rid of the `Fragment` but I thought doing that later to avoid to add more complexity to this patch

How will performance be affected by that (if at all), since I believe that the reason for using a `DocumentFragment` was that we attach the textLayer-div to the DOM upfront (and the textLayer parsing/rendering is asynchronous)?

Reply:
I'd say that the perf impact shouldn't be significant (if any) especially because we only render the text layer one time.

Reply:
Hopefully that's correct, and I suppose that the textLayer-`div` is hidden during all of this so that hopefully helps as well.

Should we perhaps also remove the mention of `DocumentFragment` from the JSDoc?
https://github.com/mozilla/pdf.js/blob/fa54a587906100eb8198de2e95d6b1e0bfa88846/src/display/text_layer.js#L30-L31

Reply:
Also, would it be feasible to have `updateTextLayer` call `setTextLayerDimensions` internally such that we no longer need to expose the latter on in the API?

Reply:
I tried to see something with the profiler but I didn't manage to so it's very likely insignificant.